### ISSUE TITLE
[BUG]: compare_models() not giving any output (stuck at status "Loading Dependencies" and library "Compiling Library")

### ISSUE THREAD

**Issue by KhalidAmine** [pycaret/pycaret#3538]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I am currently using version 3 (latest version of pycaret) and i get the following issue when using compare_models()

<img width="684" alt="Screenshot 2023-05-08 at 00 54 24" src="https://user-images.githubusercontent.com/77658006/236706474-8be34b99-4340-4ebc-9df5-08aadf6e68b7.png">

### Reproducible Example

```python
See screenshot
```

### Expected Behavior

I expect the compare_models() to give me the classification models I would like it to compare. 

### Actual Results

```python-traceback
/
```

### Installed Versions

<details>
System:
    python: 3.8.0 (v3.8.0:fa919fdf25, Oct 14 2019, 10:23:27)  [Clang 6.0 (clang-600.0.57)]
executable: [/Users/khalid-amine/Desktop/Dev/ML_toxicity/MLenv/bin/python](https://file+.vscode-resource.vscode-cdn.net/Users/khalid-amine/Desktop/Dev/ML_toxicity/MLenv/bin/python)
   machine: macOS-10.16-x86_64-i386-64bit

PyCaret required dependencies:
                 pip: 19.2.3
          setuptools: 41.2.0
             pycaret: 3.0.0
             IPython: 8.12.2
          ipywidgets: 8.0.6
                tqdm: 4.65.0
               numpy: 1.24.3
              pandas: 1.5.3
              jinja2: 3.1.2
               scipy: 1.10.1
              joblib: 1.2.0
             sklearn: 1.2.2
                pyod: 1.0.9
            imblearn: 0.10.1
   category_encoders: 2.6.0
            lightgbm: 3.3.5
               numba: 0.57.0
            requests: 2.29.0
...
           evidently: Not installed
               fugue: Not installed
           streamlit: Not installed
             prophet: Not installed
</details>

**Comment by samaelizabeth** [comment]
Had the same issue and installing sktime=0.17.0 resolved it

**Comment by KhalidAmine** [comment]
Did not solve it for me, it seems to work under the hood but not showing a table as output

<img width="774" alt="Screenshot 2023-05-08 at 10 05 15" src="https://user-images.githubusercontent.com/77658006/236770328-83547121-805c-4dba-8c6d-fa4c3ae3fc35.png">

**Comment by ngupta23** [comment]
Please share a reproducible example (code + data) so we can debug. We can not debug these kinds of issues with screenshots. Thanks!

**Comment by robertasgabrys** [comment]
compare_models() function in time series module stopped working for me too, here is an example:

https://colab.research.google.com/drive/1mJd6-8QQ7TZyX9Ub2nZbqxT2vw07B45p?usp=sharing

I tried running the pycaret time series tutorial code using airline data (https://pycaret.gitbook.io/docs/get-started/tutorials) and it did not work either: 

https://colab.research.google.com/github/pycaret/pycaret/blob/master/tutorials/Tutorial%20-%20Time%20Series%20Forecasting.ipynb

**Comment by erikjamesmason** [comment]
likewise as @robertasgabrys said - attempting locally with VS Code with PyCaret 3.0.0, specific to time series module.

**Comment by ngupta23** [comment]
This is related to the latest release of sktime. It seems to be incompatible with pycaret. Please install this version and then it works. We have pinned sktime now on master branch and the next release should prevent this issue from happening.

```
!pip install sktime==0.17.1
```

**Comment by agneev-tripathy** [comment]
> Related to #3530
> 
> This is related to the latest release of sktime. It seems to be incompatible with pycaret. Please install this version and then it works. We have pinned sktime now on master branch and the next release should prevent this issue from happening.
> 
> ```
> !pip install sktime==0.17.1
> ```
> 
> ![image](https://user-images.githubusercontent.com/33585645/237082261-339c6ec1-b5d6-4cbd-94a1-6a80c129c240.png)

I tried this solution, but still facing same problem

**Comment by ngupta23** [comment]
Please share a reproducible notebook with data so we can look further into it.

**Comment by adalseno** [comment]
Installing sktime==0.17.1 worked for me, but I have an issue with the widget (javascript error; I think it's related to tqdm):
```
[Open Browser Console for more detailed log - Double click to close this message]
Failed to load model class 'HBoxModel' from module '@jupyter-widgets/controls'
Error: Module @jupyter-widgets/controls, version ^1.5.0 is not registered, however,         2.0.0 is
    at f.loadClass (https://192.168.1.10:9999/lab/extensions/@jupyter-widgets/jupyterlab-manager/static/134.402424ef4079078b2e0e.js?v=402424ef4079078b2e0e:1:74976)
    at f.loadModelClass (https://192.168.1.10:9999/lab/extensions/@jupyter-widgets/jupyterlab-manager/static/150.1a6d6a3a0542a41bec5a.js?v=1a6d6a3a0542a41bec5a:1:10729)
    at f._make_model (https://192.168.1.10:9999/lab/extensions/@jupyter-widgets/jupyterlab-manager/static/150.1a6d6a3a0542a41bec5a.js?v=1a6d6a3a0542a41bec5a:1:7517)
    at f.new_model (https://192.168.1.10:9999/lab/extensions/@jupyter-widgets/jupyterlab-manager/static/150.1a6d6a3a0542a41bec5a.js?v=1a6d6a3a0542a41bec5a:1:5137)
    at f.handle_comm_open (https://192.168.1.10:9999/lab/extensions/@jupyter-widgets/jupyterlab-manager/static/150.1a6d6a3a0542a41bec5a.js?v=1a6d6a3a0542a41bec5a:1:3894)
    at _handleCommOpen (https://192.168.1.10:9999/lab/extensions/@jupyter-widgets/jupyterlab-manager/static/134.402424ef4079078b2e0e.js?v=402424ef4079078b2e0e:1:73392)
    at b._handleCommOpen (https://192.168.1.10:9999/static/lab/jlab_core.4d5114e25256eeddf4e4.js?v=4d5114e25256eeddf4e4:2:1039655)
    at async b._handleMessage (https://192.168.1.10:9999/static/lab/jlab_core.4d5114e25256eeddf4e4.js?v=4d5114e25256eeddf4e4:2:1041645)
```
Everything was installed in a new environment with python 3.8 and `pycaret[full]`
Is there a way to know which model pycaret is training?  The first results show almost immediately, but then it gets stuck and I don't know what model is causing the issue.

**Comment by ngupta23** [comment]
Try using sktime 0.17.0 for now. Also, would recommend not installing full. It comes with > 40 additional libraries. Only install the ones that you need.

**Comment by ngupta23** [comment]
When you run `compare_models`, it should how in the notebook which model is running at that time.

**Comment by adalseno** [comment]
On the notebook I get the widget error so I can't see any update. On vscode I get the progress bar but no hint about the model 

Inviato da Yahoo Mail su Android 
 
  Il Sab, 13 Mag, 2023 alle 18:39, Nikhil ***@***.***> ha scritto:   


When you run compare_models, it should how in the notebook which model is running at that time.

**Comment by ngupta23** [comment]
Let's try another way. Can you run compare_models with error='raise'

**Comment by adalseno** [comment]
I did it without any issues (except for the widget, which is a javascript problem and has been solved by updating ipywidgets). Now everything works fine. The problem was with auto-arima (I used 4 jobs, and it was eating about 150GB of RAM, so I excluded it, and now it works like a charm). Thank you very much.

PS I'm on Mac Os X 10.15 and there are issues with CatBoost (1.2 is not supported, and you must explicitly install 1.1.1 before installing pycaret[modules]; otherwise, it will try to install 1.2 from source and fail during compiling)
